chimneyCats/src/main/scala/io/scalaland/chimney/cats/package.scala

```diff
@@ -15,10 +18,58 @@ trait CatsImplicits extends LowPriorityImplicits {
 
   implicit def TransformerFValidatedNelSupport[E]: TransformerFSupport[ValidatedNel[E, +*]] =
     TransformerFValidatedSupport[NonEmptyList[E]](implicitly)
+
+  implicit def TransformerFIorNecSupport[E]: TransformerFSupport[IorNec[E, +*]] =
+    TransformerFIorSupport[NonEmptyChain[E]](implicitly)
+
+  implicit def TransformerFIorNelSupport[E]: TransformerFSupport[IorNel[E, +*]] =
+    TransformerFIorSupport[NonEmptyList[E]](implicitly)
+
+  implicit def TransformerFIorNesSupport[E]: TransformerFSupport[IorNes[E, +*]] =
+    TransformerFIorSupport[NonEmptySet[E]](implicitly)
 }
 
 trait LowPriorityImplicits {
 
+  implicit def TransformerFIorSupport[EE: Semigroup]: TransformerFSupport[Ior[EE, +*]] =
+    new TransformerFSupport[Ior[EE, +*]] {
+      override def pure[A](value: A): Ior[EE, A] =
+        Ior.right(value)
+
+      override def product[A, B](fa: Ior[EE, A], fb: => Ior[EE, B]): Ior[EE, (A, B)] =
+        fa.flatMap(a => fb.map(b => (a, b)))
+
+      override def map[A, B](fa: Ior[EE, A], f: A => B): Ior[EE, B] =
+        fa.map(f)
+
+      override def traverse[M, A, B](it: Iterator[A], f: A => Ior[EE, B])(implicit fac: Factory[B, M]): Ior[EE, M] = {
+        var leftSide: EE = null.asInstanceOf[EE]
+        val rightSide = fac.newBuilder
+        var isLeft = false
+
+        while (it.hasNext && !isLeft) {
+          f(it.next()) match {
+            case Ior.Left(a) =>
+              if (leftSide == null) leftSide = a
+              else leftSide = Semigroup[EE].combine(leftSide, a)
```

Comment:
Assuming early termination, you don't need to check if `leftSide == null` - it will always be true.

Reply:
So that’s true if you have Rights followed by a Left but this is not the case for Boths followed by a Left because the left side will have accumulated data and we would need to combine the data from the first Left before stopping

Reply:
Oh yes, you're right - assuming you want to return also accumulated data, not only the last Left's one.